util/reggen/ip_block.py

```diff
@@ -217,15 +290,63 @@ def from_raw(param_defaults: List[Tuple[str, str]],
             else:
                 known_cms.update({str(x): 1})
 
+        cip_id = check_int(rd.get('cip_id'), 'cip id for ' + what)
+        # In case there are multiple past revisions of this IP, always pick the
+        # newest one. Note: this revision list may be deprecated in the
+        # future.
+        version = Version('0.0.0')
+        if 'revisions' in rd:
+            for rev in check_list(rd['revisions'], what):
+                rev = check_keys(rev, 'rev item at ' + what,
+                                 list(REQUIRED_REVISIONS_FIELDS.keys()),
+                                 list(OPTIONAL_REVISIONS_FIELDS.keys()))
+                try:
+                    ver = Version(rev['version'])
+                except ValueError as err:
+                    raise RuntimeError(str(err) + ' in ' + what)
+                version = ver if ver >= version else version
+        else:
+            try:
+                version = Version(rd.get('version'))
+            except ValueError as err:
+                raise RuntimeError(str(err) + ' in ' + what)
+
+        # Add automatically generated registers in the following order:
+        # 1) CIP_ID and version
+        # 2) feature register record
+        # 3) interrupt registers
+        # 4) alert registers
+        no_auto_id = check_bool(rd.get('no_auto_id_regs', False),
+                                'no_auto_id_regs field of ' + what)
+
+        no_auto_feat = check_bool(rd.get('no_auto_feat_regs', False),
+                                  'no_auto_feat_regs field of ' + what)
+
         no_auto_intr = check_bool(rd.get('no_auto_intr_regs', not interrupts),
                                   'no_auto_intr_regs field of ' + what)
 
         no_auto_alert = check_bool(rd.get('no_auto_alert_regs', not alerts),
                                    'no_auto_alert_regs field of ' + what)
 
+        # Insert CIP_ID and version registers.
+        if not no_auto_id and cip_id is not None:
+            if cip_id not in KNOWN_CIP_IDS.keys():
+                raise RuntimeError('Unknown CIP_ID {}'.format(cip_id))
+            if KNOWN_CIP_IDS[cip_id] != name:
+                raise RuntimeError('Incorrect name {} for CIP_ID {}'
+                                   .format(name, cip_id))
+            init_block.make_id_regs(cip_id, version, what)
+
+        # TODO: this currently allocates an empty parameter block and
+        # reserves space for a parameter block that is currently empty.
+        # The plan is to extend this in the future so that CIP features/
+        # parameters can be automatically mapped into this region.
+        if not no_auto_feat:
+            init_block.make_feat_regs(what)
```

Comment:
I understand and support the decision to reserve the space now. Would it be worth documenting in the README how much space is reserved when the registers are generated?